includes/log-handlers/class-wc-log-handler-email.php

```diff
@@ -0,0 +1,117 @@
+<?php
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly
+}
+
+if ( ! class_exists( 'WC_Log_Handler' ) ) {
+	include_once( dirname( dirname( __FILE__ ) ) . '/abstracts/abstract-wc-log-handler.php' );
+}
+
+/**
+ * Handles log entries by sending an email.
+ *
+ * @class          WC_Log_Handler_Email
+ * @version        1.0.0
+ * @package        WooCommerce/Classes/Log_Handlers
+ * @category       Class
+ * @author         WooThemes
+ */
+class WC_Log_Handler_Email extends WC_Log_Handler {
+
+	/**
+	 * Stores email recipients.
+	 *
+	 * @var array
+	 * @access private
+	 */
+	private $to = array();
+
+	/**
+	 * Constructor for the logger.
+	 *
+	 * @param $args additional args. {
+	 *     Optional. @see WC_Log_Handler::__construct.
+	 *
+	 *     @type string|array $to Optional. Email or emails to recieve log messages. Default site admin.
+	 * }
+	 */
+	public function __construct( $args = array() ) {
+
+		$args = wp_parse_args( $args, array(
+			'threshold' => 'critical',
+			'to' => get_option( 'admin_email' ),
+		) );
+
+		parent::__construct( $args );
+
+		if ( is_array( $args['to'] ) ) {
+			foreach ( $args['to'] as $to ) {
+				$this->add_email( $to );
+			}
+		} else {
+			$this->add_email( $args['to'] );
+		}
+
+	}
+
+	/**
+	 * Handle a log entry.
+	 *
+	 * @param int $timestamp Log timestamp.
+	 * @param string $level emergency|alert|critical|error|warning|notice|info|debug
+	 * @param string $message Log message.
+	 * @param array $context Optional. Additional information for log handlers.
+	 *
+	 * @return bool log entry should bubble to further loggers.
+	 */
+	public function handle( $timestamp, $level, $message, $context ) {
+
+		if ( $this->should_handle( $level ) ) {
+			$subject = $this->get_subject( $timestamp, $level, $message, $context );
+			$body = $this->get_body( $timestamp, $level, $message, $context );
+			wp_mail( $this->to, $subject, $body );
```

Comment:
That is probably a good idea, under the right conditions this could send enough mail to be harmful.

Are there existing wp_mail throttling solutions?

A couple of options:

1. 1 email/request: store emails and send a summary email at some final action ([`shutdown`](https://developer.wordpress.org/reference/hooks/shutdown/)?). This avoids multiple emails per request, but is not a perfect throttling solution. Emails could be lost if execution is stopped before the action is reached and a lot of emails could still be sent if traffic is high.
1. 0-1 email/X minutes: Store emails in DB, aggregate and send them periodically. This would be a more robust email throttling solution. `wp_cron` would probably have to be involved and possibly more tables added to the db.